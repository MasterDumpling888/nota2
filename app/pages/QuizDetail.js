import React, { useState } from 'react';
import { ScrollView, View, Text, TextInput, TouchableOpacity, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import PageBox from '../components/PageBox';
import Icon from 'react-native-vector-icons/Ionicons';
import { getFontSize } from '../responsiveFont';

function QuizDetail({ route, navigation }) {
  const { quiz } = route.params;
  const [revealedAnswers, setRevealedAnswers] = useState(Array(quiz.length).fill(false));

  const toggleAnswer = (index) => {
    const newRevealedAnswers = [...revealedAnswers];
    newRevealedAnswers[index] = !newRevealedAnswers[index];
    setRevealedAnswers(newRevealedAnswers);
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <ScrollView>
        <PageBox title="Quiz" onClose={() => navigation.goBack()}>
          {quiz.map((qaPair, index) => (
            <View key={index} style={styles.questionContainer}>
              <View style={styles.question}>
                <Text style={[styles.questionAnswerText, { fontFamily: 'Raleway-SemiBold', color: 'white' }]}>Question {index + 1}: </Text>
                <Text type="default" style={[styles.questionAnswerText, { color: 'white' }]}>{qaPair.question}</Text>
              </View>
              <TextInput
                style={styles.answerInput}
                placeholder="Your answer"
                multiline
              />
              <TouchableOpacity onPress={() => toggleAnswer(index)} style={styles.revealButton}>
                <Text style={styles.revealButtonText}>
                  {revealedAnswers[index] ? 'Hide Answer' : 'Reveal Answer'}
                </Text>
                <Icon name={revealedAnswers[index] ? "eye-off-outline" : "eye-outline"} size={20} color="black" />
              </TouchableOpacity>
              {revealedAnswers[index] && (
                <View style={styles.answer}>
                  <Text style={[styles.questionAnswerText, { fontFamily: 'Raleway-SemiBold', }]}>Answer {index + 1}: </Text>
                  <Text type="default" style={styles.questionAnswerText}>{qaPair.answer}</Text>
                </View>
              )}
            </View>
          ))}
        </PageBox>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  questionContainer: {
    flex: 1,
    gap: 10,
    padding: 10,
    borderBottomWidth: 1,
    borderColor: '#C9C9C9',
    borderRadius: 5,
  },
  question: {
    borderWidth: 2,
    borderColor: '#33FD0A',
    padding: 20,
    borderRadius: 5,
    gap: 10
  },
  questionAnswerText: {
    color: 'black',
    fontSize: getFontSize(16),
    fontFamily: 'Raleway-Regular',
  },
  answerInput: {
    marginTop: 10,
    padding: 20,
    backgroundColor: '#CFFFC5',
    borderRadius: 5,
    fontSize: getFontSize(16),
    color: 'black',
    fontFamily: 'Raleway-Regular',
  },
  answer: {
    borderTopWidth: 5,
    borderTopColor: '#33FD0A',
    backgroundColor: '#33FD0A',
    padding: 20,
    borderRadius: 5,
    gap: 10
  },
  revealButton: {
    marginTop: 10,
    padding: 10,
    backgroundColor: '#32FB0A',
    borderRadius: 5,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    gap: 5,
  },
  revealButtonText: {
    color: 'black',
    fontSize: getFontSize(16),
    fontFamily: 'Raleway-Bold',
  },
  answerText: {
    marginTop: 10,
    fontSize: getFontSize(16),
    color: 'white',
    fontFamily: 'Raleway-Regular',
  },
});

export default QuizDetail;
