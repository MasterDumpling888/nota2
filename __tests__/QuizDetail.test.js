import React from 'react';
import { render } from '@testing-library/react-native';
import QuizDetail from '../app/pages/QuizDetail';

const mockRoute = {
  params: {
    quiz: [{ question: 'Sample Question?', answer: 'Sample Answer' }]
  }
};

test('renders QuizDetail correctly', () => {
  const { getByText } = render(<QuizDetail route={mockRoute} navigation={global.mockNavigation} />);
  
  expect(getByText('Question 1:')).toBeTruthy();
  expect(getByText('Sample Question?')).toBeTruthy();
});

test('QuizDetail snapshot test', () => {
  const { toJSON } = render(<QuizDetail route={mockRoute} navigation={global.mockNavigation} />);
  expect(toJSON()).toMatchSnapshot();
});