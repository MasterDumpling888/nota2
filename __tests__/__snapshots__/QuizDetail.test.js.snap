// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizDetail snapshot test 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#181818",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "backgroundColor": "#181818",
            "flex": 1,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Icon
              color="#33FD0A"
              name="chevron-back-outline"
              size={24}
              style={
                {
                  "marginBottom": 15,
                  "padding": 0,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#33FD0A",
                  "fontFamily": "Raleway-SemiBold",
                  "fontSize": 17,
                  "marginBottom": 16,
                }
              }
            >
              Quiz
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#C9C9C9",
              "borderRadius": 5,
              "flex": 1,
              "gap": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "borderColor": "#33FD0A",
                "borderRadius": 5,
                "borderWidth": 2,
                "gap": 10,
                "padding": 20,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "black",
                    "fontFamily": "Raleway-Regular",
                    "fontSize": 15,
                  },
                  {
                    "color": "white",
                    "fontFamily": "Raleway-SemiBold",
                  },
                ]
              }
            >
              Question 
              1
              : 
            </Text>
            <Text
              style={
                [
                  {
                    "color": "black",
                    "fontFamily": "Raleway-Regular",
                    "fontSize": 15,
                  },
                  {
                    "color": "white",
                  },
                ]
              }
              type="default"
            >
              Sample Question?
            </Text>
          </View>
          <TextInput
            multiline={true}
            placeholder="Your answer"
            style={
              {
                "backgroundColor": "#CFFFC5",
                "borderRadius": 5,
                "color": "black",
                "fontFamily": "Raleway-Regular",
                "fontSize": 15,
                "marginTop": 10,
                "padding": 20,
              }
            }
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#32FB0A",
                "borderRadius": 5,
                "flexDirection": "row",
                "gap": 5,
                "justifyContent": "center",
                "marginTop": 10,
                "opacity": 1,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Raleway-Bold",
                  "fontSize": 15,
                }
              }
            >
              Reveal Answer
            </Text>
            <Icon
              color="black"
              name="eye-outline"
              size={20}
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
